pkg/operator/sync.go

```diff
@@ -572,6 +572,12 @@ func (optr *Operator) syncMachineConfigController(config *renderConfig) error {
 	// new controller can roll out too.
 	// https://bugzilla.redhat.com/show_bug.cgi?id=1879099
 	cc.Annotations[daemonconsts.GeneratedByVersionAnnotationKey] = version.Raw
+
+	// add ocp release version as annotation to controller config, for use when
+	// annotating rendered configs with same.
+	optrVersion, _ := optr.vStore.Get("operator")
```

Comment:
One thing I had to dig so I can understand this better.  This value comes from the `RELEASE_IMAGE` environment variable.  Which is in turn set in the deployment's pod spec for the operator.

And the way *that* works is:

```
        env:
          - name: RELEASE_VERSION
            value: "0.0.1-snapshot"
```

Except, when a release image is generated, that value will be substituted with the OCP release version.

Which then actually means that every upgrade will result in the operator respawning, even though the code didn't change.  I hadn't realized that.

Reply:
`Which then actually means that every upgrade will result in the operator respawning`

Yeeeep. Operator does everything it would do in a "normal" (ie release involves a mco code change).